src/java/com/puppetlabs/http/client/impl/PersistentSyncHttpClient.java

```diff
@@ -26,10 +31,42 @@ private static void logAndRethrow(String msg, Throwable t) {
         throw new HttpClientException(msg, t);
     }
 
+    public Map<String, Timer> getClientMetrics(){
+        return Metrics.getClientMetrics(metricRegistry);
+    }
+
+    public Map<String, Timer> getClientMetrics(String url, Metrics.MetricType metricType) {
```

Comment:
One thing that is a little odd about having the "get"metrics APIs connected to a client object is that if you have one MetricRegistry in use by two client instances, calling getClientMetrics* on one of these clients will return metrics for requests from both clients. Looking at it from this point of view, I think this actually might be really confusing the way this is implemented and would be less confusing if the "get" metrics APIs operated on the MetricRegistry object instead.